core/src/main/java/io/temporal/samples/taskinteraction/WorkflowTaskManagerImpl.java

```diff
@@ -0,0 +1,113 @@
+/*
+ *  Copyright (c) 2020 Temporal Technologies, Inc. All Rights Reserved
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.samples.taskinteraction;
+
+import io.temporal.workflow.Workflow;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.StringTokenizer;
+
+public class WorkflowTaskManagerImpl implements WorkflowTaskManager {
+
+  private PendingTasks pendingTasks = new PendingTasks();
+
+  @Override
+  public void execute(final PendingTasks inputPendingTasks) {
+
+    initTaskList(inputPendingTasks);
+
+    Workflow.await(() -> Workflow.getInfo().isContinueAsNewSuggested());
```

Comment:
maybe im missing something here but if all tasks in task list complete 
before continueAsNewSuggested triggers this exec will running

i wonder if this can be changed to like "if we did not receive any new tasks for duration X" maybe instead?
reason is that cost of running execs on cloud so users might wanna complete this exec and use signalWithStart instead when new tasks come in? maybe im missing something so let me know

Reply:
I can add a condition so if there are no pending tasks the workflow returns/closes, WDYT @tsurdilo